addons/mail/static/tests/gif_picker/gif_picker_test.js

```diff
@@ -0,0 +1,125 @@
+/* @odoo-module */
+
+import {
+    click,
+    insertText,
+    start,
+    startServer,
+    waitUntil,
+} from "@mail/../tests/helpers/test_utils";
+import { patchUiSize, SIZES } from "../helpers/patch_ui_size";
+
+QUnit.module("gif");
+
+QUnit.test("composer should display an emoji button", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    assert.containsOnce($, "button[aria-label='Gifs']");
+});
+
+QUnit.test("Composer gif button should open the gif picker", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[aria-label='Gifs']");
+    assert.containsOnce($, ".o-discuss-gif-picker-popover");
+});
+
+QUnit.test("Searching for a gif", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[aria-label='Gifs']");
+    await insertText("input[placeholder='Search for a gif']", "search");
+    assert.containsOnce($, ".o-discuss-gif-picker-popover .fa-arrow-left");
+    await waitUntil(".o-discuss-gif", 2);
+});
+
+QUnit.test("Click on a gif category", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[aria-label='Gifs']");
+    await click("img[data-src='https://media.tenor.com/6uIlQAHIkNoAAAAM/cry.gif']");
+    assert.containsOnce($, ".o-discuss-gif-picker-popover .fa-arrow-left");
+    await waitUntil(".o-discuss-gif", 2);
+    assert.strictEqual(
+        document.querySelector("input[placeholder='Search for a gif']").value,
+        "cry"
+    );
+});
```

Comment:
Name of test is too broad, and forces us to read the test, which takes time.
If you name the test so it tells precisely what is tested, you don't necessarily have to read the test, and we waste less time in maintaining tests.